scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamOp.scala

```diff
@@ -0,0 +1,61 @@
+package com.twitter.scalding.beam_backend
+
+import com.twitter.scalding.Config
+import com.twitter.scalding.beam_backend.BeamFunctions._
+import com.twitter.scalding.typed.TypedSource
+import org.apache.beam.sdk.Pipeline
+import org.apache.beam.sdk.transforms._
+import org.apache.beam.sdk.values.PCollection
+import scala.collection.JavaConverters._
+
+sealed abstract class BeamOp[+A] {
+  import BeamOp.TransformBeamOp
+
+  def run(pipeline: Pipeline): PCollection[_ <: A]
+
+  def map[C >: A, B](f: C => B)(implicit kryoCoder: KryoCoder): BeamOp[B] =
```

Comment:
I am still pretty sure: `def map[B](f: A => B)...` works here because `Function1` is contravariant in the first parameter.